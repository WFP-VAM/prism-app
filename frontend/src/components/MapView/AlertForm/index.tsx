import {
  Box,
  Button,
  createStyles,
  TextField,
  Theme,
  Typography,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { ArrowDropDown, Notifications } from '@material-ui/icons';
import React, {
  Dispatch,
  SetStateAction,
  useCallback,
  useMemo,
  useState,
} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BoundaryLayerProps,
  LayerKey,
  WMSLayerProps,
} from '../../../config/types';
import {
  getBoundaryLayerSingleton,
  LayerDefinitions,
} from '../../../config/utils';
import { LayerData } from '../../../context/layers/layer-data';
import { layerDataSelector } from '../../../context/mapStateSlice/selectors';
import {
  AlertRequest,
  fetchApiData,
  getPrismUrl,
} from '../../../utils/analysis-utils';
import LayerDropdown from '../Layers/LayerDropdown';
import BoundaryDropdown from '../Layers/BoundaryDropdown';
import { getSelectedBoundaries } from '../../../context/mapSelectionLayerStateSlice';
import { addNotification } from '../../../context/notificationStateSlice';
import { useSafeTranslation } from '../../../i18n';

// Not fully RFC-compliant, but should filter out obviously-invalid emails.
// Source: https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
const EMAIL_REGEX: RegExp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

// This should probably be determined on a case-by-case basis,
// depending on if the downstream API has the capability.
// For now it can be permanently enabled.
const ALERT_FORM_ENABLED = true;

// TODO: Dynamic switch between local/production URLs, and consolidate this into util class
const ALERT_API_URL = 'https://prism-api.ovio.org/alerts';
// const ALERT_API_URL = 'http://localhost:80/alerts';

const boundaryLayer = getBoundaryLayerSingleton();

function AlertForm({ classes, isOpen, setOpen }: AlertFormProps) {
  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
    | LayerData<BoundaryLayerProps>
    | undefined;
  const regionsList = useSelector(getSelectedBoundaries);
  const dispatch = useDispatch();
  // form elements
  const [hazardLayerId, setHazardLayerId] = useState<LayerKey>();
  const [emailValid, setEmailValid] = useState<boolean>(false);
  const [email, setEmail] = useState('');
  const [belowThreshold, setBelowThreshold] = useState('');
  const [aboveThreshold, setAboveThreshold] = useState('');
  const [thresholdError, setThresholdError] = useState<string | null>(null);
  const [alertName, setAlertName] = useState('');
  const [alertWaiting, setAlertWaiting] = useState(false);

  const { t } = useSafeTranslation();
  const regionCodesToFeatureData: { [k: string]: object } = useMemo(() => {
    if (!boundaryLayerData) {
      // Not loaded yet. Will proceed when it is.
      return {};
    }

    return Object.fromEntries(
      boundaryLayerData.data.features
        .filter(feature => feature.properties !== null)
        .map(feature => {
          return [feature.properties?.[boundaryLayer.adminCode], feature];
        }),
    );
  }, [boundaryLayerData]);

  const generateGeoJsonForRegionNames = useCallback(() => {
    // TODO - Handle these errors properly.
    if (!boundaryLayerData) {
      throw new Error('Boundary layer data is not loaded yet.');
    }

    if (regionsList.length === 0) {
      throw new Error('Please select at least one region boundary.');
    }

    const features = regionsList.map(region => {
      return regionCodesToFeatureData[region];
    });

    // Generate a copy of admin layer data (to preserve top-level properties)
    // and replace the 'features' property with just the selected regions.
    const mutableFeatureCollection = JSON.parse(
      JSON.stringify(boundaryLayerData.data),
    );
    // eslint-disable-next-line fp/no-mutation
    mutableFeatureCollection.features = features;

    return mutableFeatureCollection;
  }, [boundaryLayerData, regionCodesToFeatureData, regionsList]);

  const onChangeEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newEmail = event.target.value;
    setEmailValid(!!newEmail.match(EMAIL_REGEX));
    setEmail(newEmail);
  };

  const onOptionChange = <T extends string>(
    setterFunc: Dispatch<SetStateAction<T>>,
  ) => (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value as T;
    setterFunc(value);
    return value;
  };

  // specially for threshold values, also does error checking
  const onThresholdOptionChange = useCallback(
    (thresholdType: 'above' | 'below') => (
      event: React.ChangeEvent<HTMLInputElement>,
    ) => {
      const setterFunc =
        thresholdType === 'above' ? setAboveThreshold : setBelowThreshold;
      const changedOption = onOptionChange(setterFunc)(event);
      // setting a value doesn't update the existing value until next render,
      // therefore we must decide whether to access the old one or the newly change one here.
      const aboveThresholdValue = parseFloat(
        thresholdType === 'above' ? changedOption : aboveThreshold,
      );
      const belowThresholdValue = parseFloat(
        thresholdType === 'below' ? changedOption : belowThreshold,
      );
      if (belowThresholdValue > aboveThresholdValue) {
        setThresholdError('Below threshold is larger than above threshold!');
      } else {
        setThresholdError(null);
      }
    },
    [aboveThreshold, belowThreshold],
  );

  const runAlertForm = useCallback(async () => {
    if (!hazardLayerId) {
      throw new Error('Layer should be selected to create alert.');
    }

    const request: AlertRequest = {
      alert_name: alertName,
      alert_config: LayerDefinitions[hazardLayerId] as WMSLayerProps,
      max: parseFloat(aboveThreshold) || undefined,
      min: parseFloat(belowThreshold) || undefined,
      zones: generateGeoJsonForRegionNames(),
      email,
      prism_url: getPrismUrl(),
    };

    setAlertWaiting(true);
    const response = await fetchApiData(ALERT_API_URL, request, dispatch);
    setAlertWaiting(false);

    if ((response as unknown) === 'Success') {
      // TODO response isn't typed correctly because fetchApiData is too strict.
      dispatch(
        addNotification({
          message: 'Success',
          type: 'success',
        }),
      );
    }
  }, [
    aboveThreshold,
    alertName,
    belowThreshold,
    dispatch,
    email,
    generateGeoJsonForRegionNames,
    hazardLayerId,
  ]);

  const renderedAlertForm = useMemo(() => {
    if (!isOpen) {
      return null;
    }
    return (
      <Box className={classes.alertFormMenu}>
        <div className={classes.newAlertFormContainer}>
          <div className={classes.alertFormOptions}>
            <Typography variant="body2">{t('Hazard Layer')}</Typography>
            <LayerDropdown
              type="wms"
              value={hazardLayerId}
              setValue={setHazardLayerId}
              className={classes.selector}
              placeholder="Choose hazard layer"
            />
          </div>
          <div className={classes.alertFormOptions}>
            <Typography variant="body2">{t('Threshold')}</Typography>
            <div className={classes.thresholdInputsContainer}>
              <TextField
                id="filled-number"
                error={!!thresholdError}
                helperText={t(thresholdError || '')}
                className={classes.numberField}
                label={t('Below')}
                InputLabelProps={{
                  style: { color: '#ffffff' },
                }}
                type="number"
                value={belowThreshold}
                onChange={onThresholdOptionChange('below')}
                variant="filled"
              />
              <TextField
                id="filled-number"
                label={t('Above')}
                className={classes.numberField}
                InputLabelProps={{
                  style: { color: '#ffffff' },
                }}
                value={aboveThreshold}
                onChange={onThresholdOptionChange('above')}
                type="number"
                variant="filled"
              />
            </div>
          </div>
          <div className={classes.alertFormOptions}>
            <Typography variant="body2">{t('Regions')}</Typography>
            <BoundaryDropdown className={classes.regionSelector} />
          </div>
          <div className={classes.alertFormOptions}>
            <TextField
              id="alert-name"
              label={t('Alert Name')}
              type="text"
              InputLabelProps={{
                style: { color: '#ffffff' },
              }}
              variant="filled"
              value={alertName}
              onChange={e => setAlertName(e.target.value)}
              fullWidth
            />
          </div>
          <div className={classes.alertFormOptions}>
            <TextField
              id="email-address"
              label={t('Email Address')}
              type="text"
              InputLabelProps={{
                style: { color: '#ffffff' },
              }}
              variant="filled"
              onChange={onChangeEmail}
              fullWidth
            />
          </div>
        </div>
        <Button
          className={classes.innerCreateAlertButton}
          onClick={runAlertForm}
          disabled={
            !hazardLayerId ||
            !!thresholdError ||
            !emailValid ||
            alertWaiting ||
            regionsList.length === 0
          }
        >
          <Typography variant="body2">{t('Create Alert')}</Typography>
        </Button>
      </Box>
    );
  }, [
    aboveThreshold,
    alertName,
    alertWaiting,
    belowThreshold,
    classes.alertFormMenu,
    classes.alertFormOptions,
    classes.innerCreateAlertButton,
    classes.newAlertFormContainer,
    classes.numberField,
    classes.regionSelector,
    classes.selector,
    classes.thresholdInputsContainer,
    emailValid,
    hazardLayerId,
    isOpen,
    onThresholdOptionChange,
    regionsList.length,
    runAlertForm,
    t,
    thresholdError,
  ]);

  if (!ALERT_FORM_ENABLED) {
    return null;
  }

  return (
    <div className={classes.alertForm}>
      <Button
        className={classes.alertTriggerButton}
        variant="contained"
        color="primary"
        onClick={() => {
          setOpen(!isOpen);
        }}
      >
        <Notifications fontSize="small" />
        <Typography variant="body2" className={classes.alertLabel}>
          {t('Create Alert')}
        </Typography>
        <ArrowDropDown fontSize="small" />
      </Button>
      {renderedAlertForm}
    </div>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    alertLabel: { marginLeft: '10px' },
    alertTriggerButton: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '3em',
      padding: theme.spacing(0.8, 2.66),
    },
    alertForm: {
      zIndex: theme.zIndex.drawer,
      textAlign: 'left',
    },
    alertFormMenu: {
      backgroundColor: theme.surfaces?.light,
      color: 'white',
      overflowX: 'hidden',
      whiteSpace: 'nowrap',
      height: 'auto',
      maxHeight: '60vh',
      width: '23vw',
      marginTop: '0.5em',
      borderRadius: '10px',
      padding: '10px',
    },
    alertFormButton: {
      height: '36px',
      marginLeft: '3px',
    },
    thresholdInputsContainer: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      gap: '1em',
      width: '100%',
    },
    alertFormOptions: {
      padding: '5px 0px',
    },
    newAlertFormContainer: {
      padding: '5px',
      marginTop: '10px',
    },
    innerCreateAlertButton: {
      backgroundColor: theme.surfaces?.dark,
      margin: '10px',
      '&:disabled': {
        opacity: '0.5',
      },
    },
    selector: {
      margin: '5px',
      width: '100%',
    },
    regionSelector: {
      minWidth: '100%',
      maxWidth: '200px',
    },
    numberField: {
      marginTop: '10px',
      width: '50%',
    },
    inputLabelRoot: {
      color: 'white',
    },
    alertFormResponseText: {
      marginLeft: '15px',
    },
  });

interface AlertFormProps extends WithStyles<typeof styles> {
  isOpen: boolean;
  setOpen: (isOpen: boolean) => void;
}

export default withStyles(styles)(AlertForm);
