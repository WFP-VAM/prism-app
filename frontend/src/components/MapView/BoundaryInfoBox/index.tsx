import React from 'react';
import {
  createStyles,
  Paper,
  Theme,
  TextField,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { useSelector } from 'react-redux';
import {
  boundsSelector,
  zoomSelector,
} from 'context/mapBoundaryInfoStateSlice';

function LocationBox({ classes }: LocationBoxProps) {
  const bounds = useSelector(boundsSelector);
  const zoom = useSelector(zoomSelector);
  const boundsStr = bounds ? JSON.stringify(bounds.toArray()) : '';
  return (
    <Paper className={classes.container}>
      <div>
        <TextField label="Boundaries" variant="standard" value={boundsStr} />
      </div>
      <div>
        <TextField
          label="Zoom"
          variant="standard"
          value={zoom ? zoom.toString() : ''}
        />
      </div>
    </Paper>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    container: {
      position: 'absolute',
      bottom: '130px',
      left: '16px',
      padding: '6px 16px',
      zIndex: theme.zIndex.modal,
      backgroundColor: theme.palette.primary.main,
    },
  });

export interface LocationBoxProps extends WithStyles<typeof styles> {}

export default withStyles(styles)(LocationBox);
