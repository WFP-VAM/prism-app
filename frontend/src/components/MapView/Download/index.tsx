import React, { useRef, useState } from 'react';
import {
  Button,
  createStyles,
  Grid,
  Hidden,
  ListItemIcon,
  ListItemText,
  MenuItem,
  Theme,
  Tooltip,
  Typography,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import Menu, { MenuProps } from '@material-ui/core/Menu';
import {
  CloudDownload,
  ArrowDropDown,
  Image,
  Description,
} from '@material-ui/icons';
import { useSelector } from 'react-redux';
import {
  layersSelector,
  mapSelector,
} from '../../../context/mapStateSlice/selectors';
import { useSafeTranslation } from '../../../i18n';
import DownloadImage from './image';
import Report from './report';
import { analysisResultSelector } from '../../../context/analysisResultStateSlice';
import { ReportType } from '../utils';

const ExportMenu = withStyles((theme: Theme) => ({
  paper: {
    border: '1px solid #d3d4d5',
    backgroundColor: theme.palette.primary.main,
  },
}))((props: MenuProps) => (
  <Menu
    elevation={0}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center',
    }}
    {...props}
  />
));

const ExportMenuItem = withStyles((theme: Theme) => ({
  root: {
    color: theme.palette.common.white,
  },
}))(MenuItem);

function Download({ classes }: DownloadProps) {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [openImage, setOpenImage] = useState(false);
  const [openReport, setOpenReport] = useState(false);
  const selectedMap = useSelector(mapSelector);
  const analysisResult = useSelector(analysisResultSelector);
  const selectedLayers = useSelector(layersSelector);
  const previewRef = useRef<HTMLCanvasElement>(null);

  const isShowingStormData = selectedLayers.some(
    ({ id }) => id === 'adamts_buffers',
  );

  const isShowingFloodData = selectedLayers.some(({ id }) =>
    id.includes('hydra_'),
  );

  const shouldShowReport = isShowingFloodData || isShowingStormData;

  const hasAnalysisData = analysisResult !== undefined;

  const tooltipText = (() => {
    if (!shouldShowReport) {
      return 'Reports can only be generated for Tropical Storms or Floods, in hazards menu';
    }
    if (!hasAnalysisData) {
      return 'Run analysis first to generate report';
    }
    return '';
  })();

  const { t } = useSafeTranslation();

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setOpenImage(false);
    setOpenReport(false);
    setAnchorEl(null);
  };

  const openModal = () => {
    if (selectedMap) {
      const activeLayers = selectedMap.getCanvas();
      const canvas = previewRef.current;
      if (canvas) {
        canvas.setAttribute('width', activeLayers.width.toString());
        canvas.setAttribute('height', activeLayers.height.toString());
        const context = canvas.getContext('2d');
        if (context) {
          context.drawImage(activeLayers, 0, 0);
        }
      }
      setOpenImage(true);
    }
    handleClose();
  };

  return (
    <Grid item>
      <Button variant="contained" color="primary" onClick={handleClick}>
        <CloudDownload fontSize="small" />
        <Hidden smDown>
          <Typography className={classes.label} variant="body2">
            {t('Export')}
          </Typography>
        </Hidden>
        <ArrowDropDown fontSize="small" />
      </Button>
      <ExportMenu
        id="export-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {/* https://v4.mui.com/components/tooltips/#disabled-elements */}
        <Tooltip placement="left" title={tooltipText}>
          <span style={{ display: 'flex' }}>
            <ExportMenuItem
              onClick={() => setOpenReport(true)}
              disabled={!shouldShowReport || !hasAnalysisData}
            >
              <ListItemIcon>
                <Description fontSize="small" htmlColor="white" />
              </ListItemIcon>
              <ListItemText primary={t('REPORT')} />
            </ExportMenuItem>
          </span>
        </Tooltip>

        <ExportMenuItem onClick={openModal}>
          <ListItemIcon>
            <Image fontSize="small" htmlColor="white" />
          </ListItemIcon>
          <ListItemText primary={t('IMAGE')} />
        </ExportMenuItem>
      </ExportMenu>
      <DownloadImage
        open={openImage}
        previewRef={previewRef}
        handleClose={handleClose}
      />
      <Report
        open={openReport}
        handleClose={handleClose}
        reportType={isShowingStormData ? ReportType.Storm : ReportType.Flood}
      />
    </Grid>
  );
}

const styles = () =>
  createStyles({
    label: {
      marginLeft: '10px',
    },
  });

export interface DownloadProps extends WithStyles<typeof styles> {}

export default withStyles(styles)(Download);
