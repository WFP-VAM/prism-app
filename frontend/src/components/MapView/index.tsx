import { memo, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  Box,
  CircularProgress,
  createStyles,
  makeStyles,
} from '@material-ui/core';
import { getDisplayBoundaryLayers } from 'config/utils';
import { addLayer } from 'context/mapStateSlice';
import {
  isLoading as areDatesLoading,
  loadAvailableDates,
} from 'context/serverStateSlice';
import { loadLayerData } from 'context/layers/layer-data';
import { mapSelector } from 'context/mapStateSlice/selectors';
import LeftPanel from './LeftPanel';
import MapComponent from './Map';
import OtherFeatures from './OtherFeatures';

/*
  reverse the order off adding layers so that the first boundary layer will be placed at the very bottom,
  to prevent other boundary layers being covered by any layers
*/
// eslint-disable-next-line fp/no-mutating-methods
const displayedBoundaryLayers = getDisplayBoundaryLayers().reverse();

const MapView = memo(() => {
  const classes = useStyles();
  // Selectors
  const datesLoading = useSelector(areDatesLoading);
  const map = useSelector(mapSelector);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadAvailableDates());

    // we must load boundary layer here for two reasons
    // 1. Stop showing two loading screens on startup - maplibre renders its children very late, so we can't rely on BoundaryLayer to load internally
    // 2. Prevent situations where a user can toggle a layer like NSO (depends on Boundaries) before Boundaries finish loading.
    displayedBoundaryLayers.forEach(l => dispatch(addLayer(l)));
    // TODO: Look into deduping this. It's not currently triggered multiple times but it could easily be.
    displayedBoundaryLayers.forEach(l =>
      dispatch(loadLayerData({ layer: l, map })),
    );
  }, [dispatch, map]);

  return (
    <Box className={classes.root}>
      <LeftPanel />
      <OtherFeatures />
      {datesLoading && (
        <div className={classes.loading}>
          <CircularProgress size={100} />
        </div>
      )}
      <MapComponent />
    </Box>
  );
});

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      height: '100%',
      width: '100%',
      position: 'relative',
    },
    loading: {
      position: 'absolute',
      height: '100%',
      width: '100%',
      backgroundColor: 'black',
      opacity: 0.75,
      zIndex: 1,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    },
  }),
);

export default MapView;
