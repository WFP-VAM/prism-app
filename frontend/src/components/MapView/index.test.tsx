import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { store } from 'context/store';
import MapView from '.';

jest.mock('./Layers/WMSLayer', () => 'mock-WMSLayer');
jest.mock('./Layers/ImpactLayer', () => 'mock-ImpactLayer');
jest.mock('./Layers/AdminLevelDataLayer', () => 'mock-AdminLevelDataLayer');
jest.mock('./Layers/BoundaryLayer', () => 'mock-BoundaryLayer');

jest.mock('./Legends', () => 'mock-Legends');
jest.mock('./DateSelector', () => 'mock-DateSelector');

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    replace: jest.fn(),
    location: {
      search: '',
    },
  }),
}));

test('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <MapView />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
