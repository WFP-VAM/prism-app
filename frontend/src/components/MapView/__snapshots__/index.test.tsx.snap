// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView renders as expected 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-root-1"
  >
    <mock-drawer
      anchor="left"
      open="true"
      paperprops="[object Object]"
      variant="persistent"
    >
      <div
        class="makeStyles-root-4 makeStyles-root-6"
      >
        <div
          class="makeStyles-tabsWrapper-5"
        >
          <div
            aria-labelledby="full-width-tab-layers"
            id="full-width-tabpanel-layers"
            role="tabpanel"
            style="display: block; flex-grow: 1; height: calc(94vh - 48px); order: -1; overflow-x: hidden;"
          >
            <div
              class="MuiBox-root MuiBox-root-7"
              style="width: 425px; display: flex; flex-direction: column; height: 100%;"
            >
              <div
                style="padding: 0.5rem;"
              >
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Rainfall"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Rainfall"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Rainfall
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Vegetation"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Vegetation"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Vegetation
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Temperature"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Temperature"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Temperature
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Tropical Storms"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Tropical Storms"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Tropical Storms
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Vulnerability"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Vulnerability"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Vulnerability
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiPaper-root MuiAccordion-root makeStyles-root-8 MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
                >
                  <div
                    aria-controls="Exposure"
                    aria-disabled="false"
                    aria-expanded="false"
                    class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-rootSummary-9"
                    id="Exposure"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content makeStyles-summaryContent-12"
                    >
                      <mock-typography
                        classes="[object Object]"
                      >
                        Exposure
                      </mock-typography>
                    </div>
                    <div
                      aria-disabled="false"
                      aria-hidden="true"
                      class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-expandIcon-11 MuiIconButton-edgeEnd"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-16"
                style="flex-grow: 1;"
              />
              <a
                href="https://github.com/WFP-VAM/prism-app/tree/sample_hash"
                rel="noopener noreferrer"
                style="color: rgb(245, 247, 248); font-size: 0.7rem; left: 1rem;"
                target="_blank"
              >
                version hash: 
                sample_hash
              </a>
            </div>
          </div>
          <div
            aria-labelledby="full-width-tab-charts"
            id="full-width-tabpanel-charts"
            role="tabpanel"
            style="display: none; flex-grow: 1; height: calc(94vh - 48px); overflow-x: auto;"
          />
          <div
            aria-labelledby="full-width-tab-analysis"
            id="full-width-tabpanel-analysis"
            role="tabpanel"
            style="display: none; flex-grow: 1; height: calc(94vh - 48px); overflow-x: auto;"
          />
          <div
            aria-labelledby="full-width-tab-alerts"
            id="full-width-tabpanel-alerts"
            role="tabpanel"
            style="display: none; flex-grow: 1; height: calc(94vh - 48px); overflow-x: auto;"
          />
          <div
            aria-labelledby="full-width-tab-none"
            id="full-width-tabpanel-none"
            role="tabpanel"
            style="display: none; flex-grow: 1; height: calc(94vh - 48px); overflow-x: auto;"
          />
        </div>
      </div>
    </mock-drawer>
    <div
      class="MuiBox-root MuiBox-root-57 makeStyles-container-55"
    >
      <div
        class="MuiBox-root MuiBox-root-58 makeStyles-optionContainer-56"
      />
    </div>
    <div
      class="makeStyles-loading-2"
    >
      <mock-circularprogress
        size="100"
      />
    </div>
    <div
      style="position: relative; width: 100%; height: 100%;"
    />
  </div>
</div>
`;
